src/extra/components/scan.py

```diff
@@ -0,0 +1,293 @@
+from datetime import timedelta
+
+import numpy as np
+from xarray import DataArray
```

Comment:
`xarray` is really slow to import and can take up to 10s of seconds especially from GPFS environments. Unlike in most other places it does look like this component will always use it, but since it's imported alongside any other component, I would prefer a lazy import.

Reply:
Sure, fixed in 7f69671.